lib/susedistribution.pm

```diff
@@ -453,8 +453,9 @@ sub init_consoles {
                 target_port => SERIAL_TERMINAL_DEFAULT_PORT});
     }
 
-    if (get_var('BACKEND', '') =~ /qemu|ikvm|generalhw/
-        || is_svirt_except_s390x)
+    if ((get_var('BACKEND', '') =~ /qemu|ikvm/
+            || (get_var('BACKEND', '') =~ /generalhw/ && get_var('GENERAL_HW_VNC_IP'))
+            || is_svirt_except_s390x))
```

Comment:
and now onto this condition :) Can we consider it the same as "has_serial_over_ssh"?

Reply:
I do not think so. It is very specific and not reused elsewhere, AFAIK.

Reply:
*sigh*, this is getting very ugly now. In general we should try to use less `if (backend …" checks in our test code and just ask the backend and console for what capabilities they provide. I unfortunately also do not have a better suggestion right now.